### ISSUE TITLE
Web service REST API calls for orders returns error when multiple websites are used

### ISSUE THREAD

**Issue by develpr** [magento/magento2#7385]
### Preconditions
1. Magento EE 2.1.2
2. PHP 7.x
3. Create two separate websites, stores, and store views (`USA Website, USA Store, USA English Store View` + Say `Canada Website, Canada Store, Canada English / French Store View`
4. Set Customer Accounts to be global (i.e. shared across websites)
5. Set base currencies to be different between different websites (so that USA can have a different product price than Canada for example).

### Steps to reproduce
1. Create customer count (`USA` or `Canada` store, doesn't matter)
2. Login as customer in USA Website / Store / View and purchase a product. In my case I checked out with a test credit card with the Braintree payment method
3. Login as the SAME customer in the `Canada Website / Store / View`, and purchase a product. In my case I purchase the same exactly product, but the Canada French version (i.e. translated into french, and with a price in CAD insted of USD).
4. Create admin integration token via API
5. Make REST API call to `GET` the orders `/rest/V1/orders?searchCriteria[pageSize]=100`

### Expected result
1. We should have gotten back a json object with all of the order information. At the very least, if I was using the API endpoint "scoped" to either CA or USA, I would have gotten either all USA or all CA orders. OR, I should have gotten both sets of orders. Something. 

### Actual result
1. Error. The error I believe is a result of Magento's API "engine" trying to cast / convert some type into what is specific in the annotations for that particular service call, but not being able to.

```[2016-11-09 20:07:29] report.CRITICAL: Exception: Notice: Array to string conversion in /var/www/mynamehere-site/vendor/magento/framework/Reflection/TypeCaster.php on line 34 in /var/www/mynamehere-site/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 /var/www/mynamehere-site/vendor/magento/framework/Reflection/TypeCaster.php(34): Magento\Framework\App\ErrorHandler->handler(8, 'Array to string...', '/var/www/mynamehere-c...', 34, Array)
#1 /var/www/mynamehere-site/vendor/magento/framework/Reflection/DataObjectProcessor.php(115): Magento\Framework\Reflection\TypeCaster->castValueToType(Array, 'string')
#2 /var/www/mynamehere-site/vendor/magento/framework/Reflection/DataObjectProcessor.php(107): Magento\Framework\Reflection\DataObjectProcessor->buildOutputDataArray(Object(Magento\Sales\Model\Order\Payment\Interceptor), '\\Magento\\Sales\\...')
#3 /var/www/mynamehere-site/vendor/magento/framework/Reflection/DataObjectProcessor.php(113): Magento\Framework\Reflection\DataObjectProcessor->buildOutputDataArray(Object(Magento\Sales\Model\Order), '\\Magento\\Sales\\...')
#4 /var/www/mynamehere-site/vendor/magento/framework/Webapi/ServiceOutputProcessor.php(107): Magento\Framework\Reflection\DataObjectProcessor->buildOutputDataArray(Object(Magento\Sales\Model\ResourceModel\Order\Collection), '\\Magento\\Sales\\...')
#5 /var/www/mynamehere-site/vendor/magento/framework/Webapi/ServiceOutputProcessor.php(59): Magento\Framework\Webapi\ServiceOutputProcessor->convertValue(Object(Magento\Sales\Model\ResourceModel\Order\Collection), '\\Magento\\Sales\\...')
#6 /var/www/mynamehere-site/vendor/magento/module-webapi/Controller/Rest.php(308): Magento\Framework\Webapi\ServiceOutputProcessor->process(Object(Magento\Sales\Model\ResourceModel\Order\Collection), 'Magento\\Sales\\A...', 'getList')
#7 /var/www/mynamehere-site/vendor/magento/module-webapi/Controller/Rest.php(216): Magento\Webapi\Controller\Rest->processApiRequest()
#8 /var/www/mynamehere-site/vendor/magento/framework/Interception/Interceptor.php(146): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#9 /var/www/mynamehere-site/var/generation/Magento/Webapi/Controller/Rest/Interceptor.php(39): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins('dispatch', Array, Array)
#10 /var/www/mynamehere-site/vendor/magento/framework/App/Http.php(135): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#11 /var/www/mynamehere-site/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#12 /var/www/mynamehere-site/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#13 {main}

Next Exception: Report ID: webapi-58238201b92b9; Message: Notice: Array to string conversion in /var/www/mynamehere-site/vendor/magento/framework/Reflection/TypeCaster.php on line 34 in /var/www/mynamehere-site/vendor/magento/framework/Webapi/ErrorProcessor.php:195
Stack trace:
#0 /var/www/mynamehere-site/vendor/magento/framework/Webapi/ErrorProcessor.php(139): Magento\Framework\Webapi\ErrorProcessor->_critical(Object(Exception))
#1 /var/www/mynamehere-site/vendor/magento/module-webapi/Controller/Rest.php(219): Magento\Framework\Webapi\ErrorProcessor->maskException(Object(Exception))
#2 /var/www/mynamehere-site/vendor/magento/framework/Interception/Interceptor.php(146): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/mynamehere-site/var/generation/Magento/Webapi/Controller/Rest/Interceptor.php(39): Magento\Webapi\Controller\Rest\Interceptor->___callPlugins('dispatch', Array, Array)
#4 /var/www/mynamehere-site/vendor/magento/framework/App/Http.php(135): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#5 /var/www/mynamehere-site/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#6 /var/www/mynamehere-site/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#7 {main} [] []
```

**Comment by veloraven** [comment]
@develpr thank you for your report.
Please report EE issues via the Support portal of your account or Partner portal if you are a partner reporting on behalf of a merchant. 
Github is intended for Community edition reports given no account management for CE users. This will allow for proper tracking of issues at the account level

**Comment by develpr** [comment]
Apparently this same issue happens across the board, and may actually be related to braintree? I really doubt this has anything to do with EE, but OK!

**Comment by YevSent** [comment]
Hi, @develpr, this is known issue and was already fixed on the develop branch (ticket MAGETWO-57426).
This fix will be available in Magento 2.1.3.

Closed as duplicate to https://github.com/magento/magento2/issues/6215.

Thanks, for reporting.
